main.py

```diff
@@ -1,13 +1,105 @@
-#!/usr/bin/env python
+import json
+import pandas as pd
+from nltk import word_tokenize
+from nltk.corpus import stopwords
+from stop_words import get_stop_words
+from nltk.wsd import lesk
+import re
 
-# Your classes or methods go here:
 
+class Writer:
 
-class Tool:
-    def my_method(self):
-        print("my_method called!")
+    """Writer data to json file. Read txt file"""
+
+    @staticmethod
+    def write_to_json(data_text):
+        with open('task.json', 'w') as json_w:
+            json.dump(data_text, json_w, indent=4)
+
+
+class Cleaner:
+
+    @staticmethod
+    def clear_data(table, col_name, pattern):
+        table[col_name].replace(pattern, ' ', regex=True)
+        return table[col_name]
+
+
+class Tokenizer:
+
+    @staticmethod
+    def delete_stop_words(table, col_name):
+        nltk_words = stopwords.words('english')
+        stop_words = get_stop_words('en')
+        stop_words.extend(nltk_words)
+        table[col_name].apply(lambda without_stopwords: ' '.join(
```

Comment:
This approach is used in all of the processing methods: modify the `table[col_name]` contents in-place and then return it. This should be avoided. You should attempt as much as possible to make your method **a pure function**. That is, it's just a pipe: data goes in, gets transformed or augmented, and then goes out.

It could be:

```python
def delete_stop_words(text):
    nltk_words = stopwords.words('english')
    stop_words = get_stop_words('en')
    stop_words.extend(nltk_words)
    table[col_name].apply(lambda without_stopwords: ' '.join(
        [word for word in without_stopwords.split() if word not in (stop_words)]))
    return table[col_name]

# calling
for token in tokens
    token_without_stop_words = delete_stop_words(token)
    # any other transformations

# merging the resulting data into a single object before writing it to the output file

data["metadata"] = ...
data["orphan_tokens"] = ...
```

Use [this comment](https://github.com/SlobodaStudio/code-style-and-composition-course-python/pull/2#discussion_r242883223) for the better understanding of what I mean.